### PR TITLE
Entity tests

### PR DESCRIPTION
**Pull request by MattTuttle**
I'm adding unit tests for Entity and found some inconsistencies in setting x/y out of order. I'm assuming this is not intended but I may be missing something.

### CODE REVIEW

File: haxepunk/Entity.hx [hunk 1]
```diff
@@ -653,22 +639,23 @@ class Entity extends Tweener
...
-		width = Reflect.getProperty(o, "width");
-		height = Reflect.getProperty(o, "height");
-
-		if (Reflect.hasField(o, "originX") && Reflect.hasField(o, "originY"))
-		{
-			originX = Reflect.getProperty(o, "originX");
-			originY = Reflect.getProperty(o, "originY");
-		}
-		else
+		var getInt = function(o:Dynamic, prop:String, defaultValue:Int=0):Int
```
Comment (bendmorris) [inline haxepunk/Entity.hx]: This could probably be an inline function.

**Comment by MattTuttle** [comment]
Any thoughts on set_x/set_y? If you set the Entity parent first it gives a different value than if you set  the instance values first.

**Comment by bendmorris** [comment]
>Any thoughts on set_x/set_y? If you set the Entity parent first it gives a different value than if you set the instance values first.

Can you give an example of what you're talking about?

**Comment by MattTuttle** [comment]
https://github.com/MattTuttle/HaxePunk/blob/6998d633bfde530833797e5f0aac404538f1022c/tests/src/haxepunk/EntityTest.hx#L307-L324

I would expect the Assert values to be equal but they are not.

**Comment by bendmorris** [comment]
Entities also have a `localX` and a `localY`; `x` and `y` are global; if you used `localX` and `localY` in the second test, the assertion values would be the same.

To me this behavior makes sense: if an entity has no parent, its `x/y` and `localX/localY` are the same and can be used interchangeably. If you set that entity's parent, it is now offset by the parent's position, which may immediately move it as well, and same if you remove its parent. However you can continue to use `x/y` to get/set its global coordinates.

**Comment by MattTuttle** [comment]
Sounds good. I wanted to make sure it wasn't unintended.

**Comment by bendmorris** [comment]
I actually just noticed that this also removed the getter/setter on Entity.width and Entity.height (hadn't pulled in a while.) Actually I find these pretty useful to trigger behavior on resize, such as containers resizing their contents or setting size constraints on an entity. Was there a reason to remove them? There's probably a minor performance impact.

**Comment by MattTuttle** [comment]
Performance and it made more sense to use the resize signal.

**Pull request merged**